src/aws_credentials.pas

```diff
@@ -19,13 +19,46 @@ interface
   //rtl
   sysutils,
   classes,
+  dateutils,
   //synapse
   synacode,
   synautil,
+  synacrypt,
+  //terceiros
+  SynCrypto,
   //aws
   aws_http;
 
+
 type
+
+  IAWSSignatureHMAC256 = interface(IInterface)
+  ['{9158D9A2-7ABA-4126-9F63-264E947AC60A}']
+    function AccessKey: string;
+    function DataStamp: string;
+    function RegionName: string;
+    function ServiceName: string;
+    function Signature: TSHA256Digest;
+  end;
+
+  { TAWSSignatureHMAC256 }
+
+  TAWSSignatureHMAC256 = class sealed(TInterfacedObject, IAWSSignatureHMAC256)
+  private
+    FAccessKey: string;
+    FDataStamp: string;
+    FRegionName: string;
+    FServiceName: string;
+  public
+    constructor Create(const sAccessKey, sDataStamp, sRegionName, sServiceName: string);
```

Comment:
@almp1 please remove these prefixes "s". Here and in all the code that you did the same.